kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,249 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        sub_domain_set=None,
+        sub_class_set=None,
+        is_valid_file: Optional[Callable[[str], bool]] = None,
+        return_domain_label: Optional[bool] = False,
+        split_train_test: Optional[bool] = False,
+        split_ratio: Optional[float] = 0.8,
+    ) -> None:
+        super(MultiDomainImageFolder, self).__init__(root, transform=transform, target_transform=target_transform)
+        domains, domain_to_idx = self._find_classes(self.root)
+        if type(sub_domain_set) == list:
+            for domain_name in sub_domain_set:
+                if domain_name not in domains:
+                    raise ValueError("Domain %s not in the image directory" % domain_name)
+            domains = sub_domain_set
+            domain_to_idx = {domain_name: i for i, domain_name in enumerate(sub_domain_set)}
+
+        classes, class_to_idx = self._find_classes(os.path.join(self.root, domains[0]))
+        if type(sub_class_set) == list:
+            for class_name in sub_class_set:
+                if class_name not in classes:
+                    raise ValueError("Class %s not in the image directory" % class_name)
+            classes = sub_class_set
+            class_to_idx = {class_name: i for i, class_name in enumerate(sub_class_set)}
+        # for domain in domains:
+        #     domain_path = os.path.join(self.root, domain)
+        #     classes_, class_to_idx_ = self._find_classes(domain_path)
+        #     if not classes == classes_:
+        #         raise ValueError("Classes for different domains are expected to be the same.")
+        samples = make_multi_domain_set(self.root, class_to_idx, domain_to_idx, extensions, is_valid_file)
+        if len(samples) == 0:
+            msg = "Found 0 files in sub-folders of: {}\n".format(self.root)
+            if extensions is not None:
+                msg += "Supported extensions are: {}".format(",".join(extensions))
+            raise RuntimeError(msg)
+
+        self.loader = loader
+        self.extensions = extensions
+
+        self.classes = classes
+        self.class_to_idx = class_to_idx
+        self.samples = samples
+        self.targets = [s[1] for s in samples]
+        self.domains = domains
+        self.domain_to_idx = domain_to_idx
+        self.domain_labels = [s[2] for s in samples]
+        self.return_domain_label = return_domain_label
+        self.split_train_test = split_train_test
+        self.split_ratio = split_ratio
+        if split_train_test:
+            self.train_idx, self.test_idx = split_by_domains(self.domain_labels, 2, [split_ratio])
+        else:
+            self.train_idx = None
+            self.test_idx = None
+
+    @staticmethod
+    def _find_classes(directory: str) -> Tuple[List[str], Dict[str, int]]:
+        """
+            Finds the class folders in a dataset.
```

Comment:
Do not call it folder. Use directory (see pytorch or https://stackoverflow.com/questions/5078676/what-is-the-difference-between-a-directory-and-a-folder)
Also, rename the class name using Dir rather than Folder.

Reply:
Oh I see pytorch using folder as well https://pytorch.org/vision/stable/_modules/torchvision/datasets/folder.html so maybe it is OK. Maybe folder is OK when files are zipped. 

Not a big issue.